contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+     *
+     */
+    function _buyComponentsForInputToken(ISetToken _setToken, uint256 _amountSetToken, bytes[] memory _quotes, IERC20 _inputToken, address _issuanceModule) internal returns (uint256 totalInputTokenSold) {
+        uint256 componentAmountBought;
+        uint256 inputTokenAmountSold;
+
+        (address[] memory components, uint256[] memory componentUnits) = getRequiredIssuanceComponents(_issuanceModule, _setToken, _amountSetToken);
+
+        uint256 inputTokenBalanceBefore = _inputToken.balanceOf(address(this));
+        for (uint256 i = 0; i < components.length; i++) {
+            address component = components[i];
+            uint256 units = componentUnits[i];
+
+            // If the component is equal to the input token we don't have to trade
+            if(component == address(_inputToken)) {
+                totalInputTokenSold = totalInputTokenSold.add(units);
+                componentAmountBought = units;
+            }
+            else {
+                uint256 componentBalanceBefore = IERC20(component).balanceOf(address(this));
+                _fillQuote(_quotes[i]);
+                uint256 componentBalanceAfter = IERC20(component).balanceOf(address(this));
+                componentAmountBought = componentBalanceAfter.sub(componentBalanceBefore);
+                require(componentAmountBought >= units, "ExchangeIssuance: UNDERBOUGHT COMPONENT");
+            }
+        }
+        uint256 inputTokenBalanceAfter = _inputToken.balanceOf(address(this));
+        totalInputTokenSold = totalInputTokenSold.add(inputTokenBalanceBefore.sub(inputTokenBalanceAfter));
+    }
+
+    /**
+     * Redeems a given list of SetToken components for given token.
+     *
+     * @param _setToken             The set token being swapped.
+     * @param _amountSetToken       The amount of set token being swapped.
+     * @param _swaps                An array containing ZeroExSwap swaps.
+     * @param _outputToken          The token for which to sell the index components
+     *
+     * @return totalOutputTokenBought  Total amount of output token received after liquidating all SetToken components
+     */
+    function _sellComponentsForOutputToken(ISetToken _setToken, uint256 _amountSetToken, bytes[] memory _swaps, IERC20 _outputToken, address _issuanceModule)
+        internal
+        returns (uint256 totalOutputTokenBought)
+    {
+        (address[] memory components, uint256[] memory componentUnits) = getRequiredRedemptionComponents(_issuanceModule, _setToken, _amountSetToken);
+        uint256 outputTokenBalanceBefore = _outputToken.balanceOf(address(this));
+        for (uint256 i = 0; i < _swaps.length; i++) {
+            uint256 maxAmountSell = componentUnits[i];
+
+            uint256 outputTokenAmountBought;
+            uint256 componentAmountSold;
+
+            // If the component is equal to the input token we don't have to trade
+            if(components[i] == address(_outputToken)) {
```

Comment:
The logic inside this `if` block is not getting hit by the unit tests.

Reply:
Good catch. I added a test case where the output token is one of the components to cover this.